src/compas_rhino/scene/meshobject.py

```diff
@@ -176,20 +176,33 @@ def draw(self, color=None, vertexcolors=None, facecolors=None, disjoint=False):
         # when drawing the mesh as individual components (vertices, edges, faces)
         # colors have to be provided as dicts that map colors to specific components
 
-        color = Color.coerce(color) or self.color
-        attr = attributes(name=self.mesh.name, color=color, layer=self.layer)  # type: ignore
+        self._guids = []
 
-        geometry = mesh_to_rhino(
-            self.mesh,
-            color=color,
-            vertexcolors=vertexcolors,
-            facecolors=facecolors,
-            disjoint=disjoint,
-        )
+        if self.show_faces is True:
+            color = Color.coerce(color) or self.color
+            attr = attributes(name=self.mesh.name, color=color, layer=self.layer)  # type: ignore
 
-        geometry.Transform(transformation_to_rhino(self.worldtransformation))
+            geometry = mesh_to_rhino(
+                self.mesh,
+                color=color,
+                vertexcolors=vertexcolors,
+                facecolors=facecolors,
+                disjoint=disjoint,
```

Comment:
Should we remove these kwargs from `draw` itself, and only allow to configure them in `scene.add()` ?